feature/song/info/src/main/java/ru/stersh/youamp/feature/song/info/ui/SongInfoScreen.kt

```diff
@@ -26,72 +28,59 @@ import androidx.compose.material3.Text
 import androidx.compose.runtime.Composable
 import androidx.compose.runtime.LaunchedEffect
 import androidx.compose.runtime.getValue
-import androidx.compose.runtime.mutableStateOf
 import androidx.compose.runtime.remember
-import androidx.compose.runtime.rememberCoroutineScope
-import androidx.compose.runtime.setValue
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.draw.clip
 import androidx.compose.ui.graphics.vector.ImageVector
 import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.unit.dp
-import kotlinx.coroutines.launch
-import org.koin.compose.koinInject
-import ru.stersh.youamp.core.api.provider.ApiProvider
+import androidx.lifecycle.ViewModelStoreOwner
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import org.koin.androidx.compose.koinViewModel
 import ru.stersh.youamp.core.ui.Artwork
 import ru.stersh.youamp.core.ui.SkeletonLayout
 import ru.stersh.youamp.core.ui.YouampPlayerTheme
 import ru.stersh.youamp.feature.song.info.R
-import ru.stersh.youamp.shared.player.queue.AudioSource
-import ru.stersh.youamp.shared.player.queue.PlayerQueueAudioSourceManager
 
 
 @Composable
 fun SongInfoScreen(
     id: String,
+    viewModelStoreOwner: ViewModelStoreOwner,
```

Comment:
ViewModelStoreOwner no needed, this changes lifecycle of ViewModel on Activity lifecycle, this not a good practice.

Reply:
I don't think it makes any difference in this case. The SongInfoScreen for the Album, Favorites, and Search screens is located at the top activity level, outside of NavHost. But I delete it.